promq/autocomplete/earley/luthor.go

```diff
@@ -219,6 +267,28 @@ func mapParserItemTypeToTokhanType(item promql.Item) TokenType {
 
 // need to explicitly extract this function since it's private
 // in prometheus 2.16
```

Comment:
Yeah... it's so annoying... I was thinking we may just want to copy them over locally..

Reply:
We copied two function and for now is enough?